java-checks/src/main/java/org/sonar/java/checks/serialization/SerializableFieldInSerializableClassCheck.java

```diff
@@ -168,4 +225,30 @@ private static boolean implementsSerializable(@Nullable Type type) {
     return false;
   }
 
+  private static final class AssignmentsVisitor extends BaseTreeVisitor {
+    private final Set<String> fields;
+    private final Set<String> usedFields = new HashSet<>();
+
+    public AssignmentsVisitor(Set<String> fields) {
+      this.fields = fields;
+    }
+
+    @Override
+    public void visitAssignmentExpression(AssignmentExpressionTree tree) {
+      if (ExpressionUtils.isSelectOnThisOrSuper(tree)) {
+        MemberSelectExpressionTree select = ((MemberSelectExpressionTree) tree.variable());
+        usedFields.add(select.identifier().name());
+      } else if (tree.variable().is(Tree.Kind.IDENTIFIER)) {
+        IdentifierTree identifier = ((IdentifierTree) tree.variable());
+        if (fields.contains(identifier.name())) {
```

Comment:
Do we really need this check? If we collect any variable assigned, it should work fine, the set returned will just contain unrelated names.

Reply:
It was a small space opitimization. As far as we return Set, should not be a big deal regarding performance